runtime/tests/interpreter/reference_test.go

```diff
@@ -1147,6 +1147,180 @@ func TestInterpretResourceReferenceInvalidationOnMove(t *testing.T) {
 		RequireError(t, err)
 		require.ErrorAs(t, err, &interpreter.InvalidatedResourceReferenceError{})
 	})
+
+	t.Run("nested resource in composite", func(t *testing.T) {
```

Comment:
I'm 99% sure the attachment equivalent of this will just work out of the box, since they are implemented the same way as nested resource fields, but just to double check, can you add another test here for the same behavior, but where `Baz` is an attachment for `Bar`?

Reply:
[This test ](https://github.com/onflow/cadence/blob/1a818a0d4a2bb36cfbc2d160dcb5b849c960bcd2/runtime/attachments_test.go#L149) already covers that path, so I didn't add anything new.

Reply:
That test was already failing before this PR with a different error. The following did not previously fail on `stable-cadence`, so I'd like to make sure that it fails properly after this change:

```
t.Parallel()

inter := parseCheckAndInterpret(t, `
          var ref: &A? = nil
          resource R {}
          attachment A for R {
              let id: Int
              fun storeSelf() {
                  ref = self
              }  
              init() {
                  self.id = 3
              }
          }
          pub fun main(): @R? {
              var r <- attach A() to <- create R()
              r[A]!.storeSelf()
              return <- r
          }
          pub fun getRef(): Int {
              return ref!.id
          }
      `,
)

_, err := inter.Invoke("main")
require.NoError(t, err)

_, err = inter.Invoke("getRef")
require.NoError(t, err)
```

Reply:
This one is also already there: https://github.com/onflow/cadence/blob/supun/sync-stable-cadence/runtime/tests/interpreter/attachments_test.go#L1754
It is now correctly failing once the master was merged to stable-cadence.

Reply:
Oh perfect thanks